internal/internal_worker_base.go

```diff
@@ -310,6 +310,34 @@ func (bw *baseWorker) runPoller() {
 	}
 }
 
+func (bw *baseWorker) tryReserveSlot() bool {
+	if !bw.isWorkerStarted || bw.isStop() {
```

Comment:
A bit scared of accessing this field in a non-thread-safe way, but probably ok since only mutated on start/stop

Reply:
@cretz @Quinn-With-Two-Ns Yes, I've found a race here on the `roadrunner-temporal` tests, on the one where we start and then stop temporal workers from the different goroutines, but protected by mutex from our side. Here is the stack trace of the race:
 
![image](https://github.com/temporalio/sdk-go/assets/8040338/1294f888-617c-4b88-aab0-7b0fd234ab0a)

Reply:
@rustatian can you link the `roadrunner-temporal` test failure? I'll look into it

Reply:
> @rustatian can you link the `roadrunner-temporal` test failure? I'll look into it

https://github.com/roadrunner-server/rr-e2e-tests/actions/runs/5790984816/job/15695006300#step:10:963